### PR TITLE
[ISSUE #6728] Compute the confirmOffset without considering new connections

### LINKED ISSUE #6728: [Bug] The confirmOffset may be set to -1 due to the new connection
### Before Creating the Bug Report

- [X] I found a bug, not just asking a question, which should be created in [GitHub Discussions](https://github.com/apache/rocketmq/discussions).

- [X] I have searched the [GitHub Issues](https://github.com/apache/rocketmq/issues) and [GitHub Discussions](https://github.com/apache/rocketmq/discussions)  of this repository and believe that this is not a duplicate.

- [X] I have confirmed that this bug belongs to the current repository, not other repositories of RocketMQ.

### Runtime platform environment

OS: centOS

### RocketMQ version

5.1.0

### JDK Version

_No response_

### Describe the Bug

The confirmOffset may be set to -1 due to the new connection.

image

If a connection is inited, but has yet been TRANSFER state, then the slaveAckOffset will be -1.
It will make the newConfirmOffset be a wrong value, which is dangerous.

### Steps to Reproduce

add a log, then you will see the confirmOffset may be -1 value.

### What Did You Expect to See?

Compute the confirmOffset without those new connections.

### What Did You See Instead?

The confirmOffset was set to -1.

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by GenerousMan**
[ISSUE #6728] Compute the confirmOffset without considering new connections.

### Which Issue(s) This PR Fixes

Fixes #6728 

### Brief Description

1. When calculate the newConfirmOffset, skip those connections with -1 ackOffset value.
2. Add a new method, getConfirmOffsetDirectly(). This method will only return the confirmOffset without checking and re-computing. Because in some cases (If set confirmOffset wrongly) endless calling will occur between the getConfirmOffset() and computeConfirmOffset().

### CODE REVIEW

File: store/src/main/java/org/apache/rocketmq/store/ha/autoswitch/AutoSwitchHAService.java [hunk 1]
```diff
@@ -421,14 +421,21 @@ public long computeConfirmOffset() {
...
         for (HAConnection connection : this.connectionList) {
             final Long slaveId = ((AutoSwitchHAConnection) connection).getSlaveId();
             if (currentSyncStateSet.contains(slaveId)) {
-                newConfirmOffset = Math.min(newConfirmOffset, connection.getSlaveAckOffset());
+                long slaveAckOffset = connection.getSlaveAckOffset();
+                if (slaveAckOffset <= 0) {
+                    // Slave's connection is just inited, the ack hasn't been calculated.
+                    // So skip this ackOffset.
+                    continue;
+                }
```
Comment (echooymxq) [inline store/src/main/java/org/apache/rocketmq/store/ha/autoswitch/AutoSwitchHAService.java:437]: if (currentSyncStateSet.contains(slaveId) && connection.getSlaveAckOffset() > 0) {
 ...
}

**Review by echooymxq — COMMENTED** [review]
currentSyncStateSet.contains(brokerAddr) && connection.getSlaveAckOffset() > 0

File: store/src/main/java/org/apache/rocketmq/store/ha/autoswitch/AutoSwitchHAService.java (hunk 1 above)
Comment (GenerousMan) [reply store/src/main/java/org/apache/rocketmq/store/ha/autoswitch/AutoSwitchHAService.java]: Yes, I have optimized this logic as you said.

**Review by fuyou001 — APPROVED** [review]
LGTM

**Review by mxsm — APPROVED** [review]
LGTM~

**Review by RongtongJin — APPROVED** [review]

**Pull request merged**